src/components/coin-icon/FastRequestVendorLogoIcon.tsx

```diff
@@ -0,0 +1,119 @@
+import React, { useCallback, useMemo, useState } from 'react';
+import { StyleSheet, View } from 'react-native';
+import { initials } from '../../utils';
+import ChainBadge from '../coin-icon/ChainBadge';
+import { Text } from '../text';
+import { CoinIconSize } from './CoinIcon';
+import { ImgixImage } from '@rainbow-me/images';
+import { position } from '@rainbow-me/styles';
+import { ThemeContextProps, useTheme } from '@rainbow-me/theme';
+import ShadowStack from 'react-native-shadow-stack';
+
+const RVLIBorderRadius = 16.25;
+const RVLIShadows = (colors: ThemeContextProps['colors']) => ({
+  default: [[0, 4, 12, colors.shadow, 0.06]],
+  large: [[0, 6, 10, colors.shadow, 0.14]],
+  none: [[0, 0, 0, colors.transparent, 0]],
+});
+
+const cx = StyleSheet.create({
+  container: {
+    alignItems: 'center',
+    justifyContent: 'center',
+  },
+  imgix: {
+    ...position.sizeAsObject('100%'),
+  },
+});
+
+export default React.memo(function RequestVendorLogoIcon({
+  backgroundColor,
+  badgeYPosition = 14,
+  borderRadius = RVLIBorderRadius,
+  dappName,
+  imageUrl,
+  noShadow,
+  shouldPrioritizeImageLoading,
+  showLargeShadow,
+  size = CoinIconSize,
+  network,
+}: {
+  backgroundColor?: string;
+  badgeYPosition?: number;
+  borderRadius?: number;
+  dappName: string;
+  imageUrl: string;
+  noShadow?: boolean;
+  shouldPrioritizeImageLoading?: boolean;
+  showLargeShadow?: boolean;
+  size?: number;
+  network?: string;
+}) {
+  const [error, setError] = useState(false);
+  const { colors } = useTheme();
+
+  const onError = useCallback(() => setError(true), [setError]);
+
+  // When dapps have no icon the bgColor provided to us is transparent.
+  // Having a transparent background breaks our UI, so we instead show a background
+  // color of white.
+  const bgColor =
+    backgroundColor === 'transparent'
+      ? colors.white
+      : backgroundColor || colors.dark;
```

Comment:
Could you use nullish coalescing there instead?

Reply:
Also a question is there any reason why we would prefer to use `||` instead of `??` in any case that involves situation where we want to add a "backup" value in case sth is nulilish?

Reply:
Yea in this case the only difference I see would be if backgroundColor was empty string.